File: complex.java
import java.util.Scanner;

class Complex {
    private double real;
    private double imaginary;

    public Complex(double r, double i) {
        real = r;
        imaginary = i;
    }

    public Complex add(Complex c) {
        double newReal = real + c.real;
        double newImaginary = imaginary + c.imaginary;
        return new Complex(newReal, newImaginary);
    }

    public void display() {
   
        System.out.println(real + " + " + imaginary + "i");
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the real and imaginary parts of the first complex number: ");
        double real1 = scanner.nextDouble();
        double imaginary1 = scanner.nextDouble();

        System.out.print("Enter the real and imaginary parts of the second complex number: ");
        double real2 = scanner.nextDouble();
        double imaginary2 = scanner.nextDouble();

        Complex complex1 = new Complex(real1, imaginary1);
        Complex complex2 = new Complex(real2, imaginary2);

        Complex sum = complex1.add(complex2);

        System.out.println("Complex Number 1: ");
        complex1.display();

        System.out.println("Complex Number 2: ");
        complex2.display();

        System.out.println("Sum of Complex Numbers: ");
        sum.display();
    }
}

File: complex.cpp
#include <iostream>
using namespace std;

class Complex {
private:
    double real;
    double imag;

public:
    Complex(double r, double i) : real(r), imag(i) {}

    Complex operator+( Complex other) {
        return Complex(real + other.real, imag + other.imag);
    }

    void displayComplex() {
        cout << real << " + " << imag << "i";
    }
};

int main() {
    double real1, imag1, real2, imag2;
    cout << "Enter the real and imaginary parts of the first complex number: ";
    cin >> real1 >> imag1;

    cout << "Enter the real and imaginary parts of the second complex number: ";
    cin >> real2 >> imag2;

    Complex complex1(real1, imag1);
    Complex complex2(real2, imag2);

    Complex sum = complex1 + complex2;

    cout << "Complex Number 1: ";
    complex1.displayComplex();
    cout << endl;

    cout << "Complex Number 2: ";
    complex2.displayComplex();
    cout << endl;

    cout << "Sum of Complex Numbers: ";
    sum.displayComplex();
    cout << endl;

    return 0;
}
